mirrors_countme/totals.py

```diff
@@ -118,38 +118,110 @@ class RawDB(SQLiteReader):
 
+    @property
+    def mintime(self):
+        return self.mintime_countme
+
+    @property
+    def maxtime(self):
+        return self.maxtime_countme
+
     def complete_weeks(self):
         """Return a range(startweek, provweek) that covers (valid + complete)
         weeknums contained in this database. The database may contain some
         data for `provweek`, but since it's provisional/incomplete it's
         outside the range."""
+        if self.mintime is None:
+            return []
         # startweek can't be earlier than the first week of data
         startweek = max(weeknum(self.mintime), COUNTME_START_WEEKNUM)
         # A week is provisional until the LOG_JITTER_WINDOW expires, so once
         # tsmax minus LOG_JITTER_WINDOW ticks over into a new weeknum, that
         # weeknum is the provisional one. So...
-        provweek = weeknum(self.maxtime - LOG_JITTER_WINDOW)
+        provweek = max(weeknum(self.maxtime - LOG_JITTER_WINDOW), COUNTME_START_WEEKNUM)
         return range(startweek, provweek)
 
-    def week_count(self, weeknum):
+    def week_iter(self, weeknum, select: tuple | list):
+        item_select = ",".join(select)
         start_ts = weeknum * WEEK_LEN + COUNTME_EPOCH
         end_ts = start_ts + WEEK_LEN
-        cursor = self._connection.execute(
-            f"SELECT COUNT(*)"
+        return self._connection.execute(
+            f"SELECT {item_select}"
             f" FROM {self._tablename}"
-            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts}"
+            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts} AND sys_age >= 0"
         )
+
+    def week_count(self, weeknum):
+        cursor = self.week_iter(weeknum, ("COUNT(*)",))
         return cursor.fetchone()[0]
 
+
+class RawDBU(RawDB):
+    def __init__(self, fp, **kwargs):
+        super().__init__(fp, **kwargs)
+
+    @property
+    def mintime(self):
+        return self.mintime_unique
+
+    @property
+    def maxtime(self):
+        return self.maxtime_unique
+
     def week_iter(self, weeknum, select: tuple | list):
-        item_select = ",".join(select)
+        start_ts = weeknum * WEEK_LEN + COUNTME_EPOCH
+        return SplitWeekDays(self, start_ts, select)
+
+    def week_count(self, weeknum):
         start_ts = weeknum * WEEK_LEN + COUNTME_EPOCH
         end_ts = start_ts + WEEK_LEN
-        return self._connection.execute(
-            f"SELECT {item_select}"
+
+        # So this is a "problem" ... we could do a GROUP BY as we do in
+        # SplitWeekDays(), but that is basically doing all the same work.
+        # Just counting the rows gets us to roughly 8:1 ... so go with that.
+        cursor = self._connection.execute(
+            f"SELECT COUNT(*)"
             f" FROM {self._tablename}"
-            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts}"
+            f" WHERE timestamp >= {start_ts} AND timestamp < {end_ts} AND sys_age < 0"
```

Comment:
Not sure about this – shouldn’t unique IP statistics be about all log entries (including those with `countme`?